src/main/kotlin/dev/gtuk/diga/DigaService.kt

```diff
@@ -64,18 +65,20 @@ class DigaService(private val appConfig: AppConfig) {
         this.apiClient = DigaApiClient(apiClientSettings, digaInformation)
     }
 
-    @Throws(ValidationException::class)
+    @Throws(ValidationException::class, TestCodesDisabledException::class)
     fun verify(code: String): ValidationResponse {
         val isTestCode = Utils.isTestCode(code)
 
         val response: DigaCodeValidationResponse = try {
             if (isTestCode && appConfig.disableTestcodes) {
-                throw CodeValidationException("Testcodes are not allowed")
+                throw TestCodesDisabledException("Testcodes are not allowed")
             } else if (isTestCode) {
                 this.apiClient.sendTestCodeValidationRequest(Utils.getDigaTestCode(code), appConfig.testInsurance)
             } else {
                 this.apiClient.validateDigaCode(code)
             }
+        } catch (e: TestCodesDisabledException) {
+            throw TestCodesDisabledException(e.message.toString())
```

Comment:
Instead of throwing a new instance with `throw TestCodesDisabledException` you could just do `throw e`. This way we preserve all the original information

Reply:
Ah of course. Will fix.